src/sdk/sdk.ts

```diff
@@ -136,11 +130,16 @@ export class PrimeSdk {
     return this.etherspotWallet.getCounterFactualAddress();
   }
 
-  async estimate(gasDetails?: TransactionGasInfoForUserOp) {
+  async estimate(paymasterDetails?: PaymasterApi, gasDetails?: TransactionGasInfoForUserOp) {
     if (this.userOpsBatch.to.length < 1) {
       throw new Error("cannot sign empty transaction batch");
     }
 
+    if (paymasterDetails?.url) {
+      const paymasterAPI = new VerifyingPaymasterAPI(paymasterDetails.url, Networks[this.chainId].contracts.entryPoint, paymasterDetails.context ?? {}, paymasterDetails.api_key, this.chainId)
```

Comment:
Does this code work with both stackup and pimlico paymasters?

Reply:
As discussed, this will be tested and make the necessary changes in a separate ticket